File: myRecipesBookBE/src/main/java/com/ceylan/myRecipesBook/entity/request/create/CreateRecipeRequest.java
package com.ceylan.myRecipesBook.entity.request.create;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CreateRecipeRequest {

    private String name;
    private int categoryId;
    private String description;
    private String imagePath;
    private int userId;
}

File: myRecipeBookFE/src/app/models/createRecipeRequest.ts
export class CreateRecipeRequest {
  constructor(
    public name: string,
    public categoryId: number,
    public description: string,
    public imagePath: string,
    public userId: number
  ) {
    this.name = name;
    this.categoryId = categoryId;
    this.description = description;
    this.imagePath = imagePath;
    this.userId = userId;
  }
}
